<launch>
	<node pkg="carl_navigation" type="virtual_fence" name="virtual_fence" output="screen" respawn="true">
		<param name="fence_x" type="double" value="0.35" />
		<param name="fence_y" type="double" value="0.35" />
		<param name="fence_width" type="double" value="4.1" />
		<param name="fence_height" type="double" value="6.0" />
		<remap from="/nav_goal" to="/move_base_simple/goal" />		
	</node>
	
	<node pkg="amcl" type="amcl" name="amcl" output="screen">
	  	<param name="min_particles" value="1000"/>
	  	<param name="max_particles" value="5000"/>
	  	<param name="kld_err" value="0.95"/>
	  	<param name="kld_z" value="0.99"/>
	  	<param name="update_min_d" value="0.025"/>
	  	<param name="update_min_a" value="0.025"/>
	  	<param name="resample_interval" value="1"/>
	  	<param name="odom_model_type" value="diff"/>
	  	<param name="gui_publish_rate" value="10.0"/>
	  	<param name="laser_max_beams" value="30"/>
	  	<param name="odom_alpha1" value="5.0"/>
	  	<param name="odom_alpha2" value="4.0"/>
	  	<param name="odom_alpha3" value="4.0"/>
	  	<param name="odom_alpha4" value="4.0"/>
	  	<param name="odom_alpha5" value="0.0"/>
	  	<param name="laser_z_hit" value="0.2"/>
	  	<param name="laser_z_short" value="0.05"/>
	  	<param name="laser_z_max" value="0.05"/>
	  	<param name="laser_z_rand" value="0.8"/>
	  	<param name="laser_sigma_hit" value="0.2"/>
	  	<param name="laser_lambda_short" value="0.1"/>
	  	<param name="laser_lambda_short" value="0.1"/>
	  	<param name="laser_model_type" value="likelihood_field"/>
	  	<param name="laser_likelihood_max_dist" value="20.0"/>
	  	<param name="odom_frame_id" value="odom"/>
	  	<param name="base_frame_id" value="base_footprint"/>
	  	<param name="transform_tolerance" value="1.0"/>
	  	<param name="recovery_alpha_slow" value="0.0"/>
  		<param name="recovery_alpha_fast" value="0.0"/>
<!--        <param name="use_map_topic" value="true" /> -->
<!--        <param name="first_map_only" value="false"/> -->
<!--        <param name="save_pose_rate" value="30" /> -->
<!--        <remap from="/map" to="/markers_to_map/ar_localization_map"/>  -->
	</node>

	<node name="move_base" pkg="move_base" type="move_base">
    	<rosparam file="$(find carl_navigation)/cfg/costmap_common_params.yaml" command="load" ns="global_costmap" />
    	<rosparam file="$(find carl_navigation)/cfg/costmap_common_params.yaml" command="load" ns="local_costmap" />
		<rosparam file="$(find carl_navigation)/cfg/local_costmap_params.yaml" command="load" />
     	<rosparam file="$(find carl_navigation)/cfg/global_costmap_params.yaml" command="load" />
      	<rosparam file="$(find carl_navigation)/cfg/base_local_planner_params.yaml" command="load" />
		<param name="controller_frequency" value="25.0" />
		<param name="planner_patience" value="7.0" />
		<param name="controller_patience" value="7.0" />
		<param name="recovery_behavior_enabled" value="true" />
		<param name="clearing_rotation_allowed" value="true" />
		<param name="prune_plan" value="false" />
 		<param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
		<param name="planner_frequency" value="1.2" />
	</node>
	
</launch>
